### PR TITLE
DER-99 - Complete reconciliation of exercise terms/date in Options

### PR DESCRIPTION
**Pull request by ElisaKendall**
## Description

This resolution includes additional clean-up of the options ontology and content related to it.  We have (1) eliminated the concepts related to novated contract, which will be generalized in FND, (2) eliminated the references to exercise conventions from bonds, (3) added a property and direct reference to exercise style in options, (4) eliminated an unused import in FuturesAndForwards and updated the DER all file to address a missing reference, (5) added the concept of an exchange participant in the markets ontology and used that instead of  'options exchange participant' in options, and (6) revised the intersection of exchange participant and option issuer with a union, all in preparation for releasing the options ontology (which is very close).

Fixes: #1529 / DER-99 / DER-64

## Checklist:

- [x] I'm familiar with the [FIBO developer quide](https://github.com/edmcouncil/fibo/blob/master/CONTRIBUTING.md#contributing-to-the-fibo-code). My contribution meets all the requirements described there.
- [x] My contribution follows the [principles of best practices for FIBO](https://github.com/edmcouncil/fibo/blob/master/ONTOLOGY_GUIDE.md).
- [x] My changes have been reconciled with latest master and no merge conflicts remain.
- [x] This PR is related to exactly one issue. The issue is referenced by using a GitHub keyword such as "fixes", "closes", or "resolves".
- [x] Hygiene tests have been applied by a PR with "(WIP)" in title.
- [x] The issue has been tested locally using a reasoner (for ontology changes).

### CODE REVIEW

**Review by tahoeblue — APPROVED** [review]

**Review by jfgemski — APPROVED** [review]

File: FBC/FunctionalEntities/Markets.rdf [hunk 1]
```diff
@@ -154,6 +159,36 @@ The securities traded on a stock exchange include: shares issued by companies, u
 		<fibo-fnd-utl-av:synonym>market</fibo-fnd-utl-av:synonym>
 	</owl:Class>
 	
+	<owl:Class rdf:about="&fibo-fbc-fct-mkt;ExchangeParticipant">
+		<rdfs:subClassOf rdf:resource="&fibo-fbc-pas-fpas;RegisteredAgent"/>
+		<rdfs:subClassOf rdf:resource="&fibo-fnd-org-org;OrganizationMember"/>
```
Comment (rivettp) [inline FBC/FunctionalEntities/Markets.rdf:164]: not sure why subClassOf organizationMember: does not seem right.

File: DER/DerivativesContracts/Options.rdf [hunk 2]
```diff
@@ -189,57 +186,26 @@
 		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">The seller (issuer) of the call option assumes the obligation of delivering the assets specified should the buyer exercise the option. The buyer has the option, but not the obligation, to purchase the underlying asset.</fibo-fnd-utl-av:explanatoryNote>
 	</owl:Class>
 	
-	<owl:Class rdf:about="&fibo-der-der-opt;EquityOption">
-		<rdfs:subClassOf rdf:resource="&fibo-der-sbd-sbd;EquityDerivative"/>
+	<owl:Class rdf:about="&fibo-der-der-opt;CombinationOption">
 		<rdfs:subClassOf rdf:resource="&fibo-fbc-fi-fi;Option"/>
```
Comment (rivettp) [inline DER/DerivativesContracts/Options.rdf:190]: Not sure the combination is itself an Option - it does not have a (single) strike price, expiration date, instrument identifier etc

File: DER/DerivativesContracts/Options.rdf [hunk 3]
```diff
@@ -189,57 +186,26 @@
...
-		</rdfs:subClassOf>
+	<owl:Class rdf:about="&fibo-der-der-opt;EquityOption">
+		<rdfs:subClassOf rdf:resource="&fibo-der-der-opt;VanillaOption"/>
+		<rdfs:subClassOf rdf:resource="&fibo-der-sbd-sbd;EquityDerivative"/>
 		<rdfs:subClassOf>
 			<owl:Restriction>
-				<owl:onProperty rdf:resource="&fibo-fnd-agr-ctr;hasCounterparty"/>
-				<owl:someValuesFrom rdf:resource="&fibo-der-der-opt;TradedOptionsCounterparty"/>
+				<owl:onProperty rdf:resource="&fibo-der-der-opt;hasExerciseStyle"/>
+				<owl:hasValue rdf:resource="&fibo-sec-dbt-ex;AmericanExerciseConvention"/>
```
Comment (rivettp) [inline DER/DerivativesContracts/Options.rdf:202]: not sure why this restriction - is it true of an option on a European exchange?

File: DER/DerivativesContracts/Options.rdf [hunk 4]
```diff
@@ -607,6 +522,52 @@
...
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fbc-fi-ip;hasPriceDeterminationMethod"/>
+				<owl:someValuesFrom rdf:resource="&fibo-fbc-fi-ip;PriceDeterminationMethod"/>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-agr-ctr;hasContractDuration"/>
```
Comment (rivettp) [inline DER/DerivativesContracts/Options.rdf:541]: no this does not seem right. The duration will continually vary driven by expiration date - it's not part of the contract at all

File: DER/DerivativesContracts/Options.rdf [hunk 5]
```diff
@@ -607,6 +522,52 @@
 		<skos:definition xml:lang="en">seller of the underlying asset in an option transaction</skos:definition>
 	</owl:Class>
 	
+	<owl:Class rdf:about="&fibo-der-der-opt;VanillaOption">
+		<rdfs:subClassOf rdf:resource="&fibo-fbc-fi-fi;Option"/>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fbc-dae-gty;hasGuarantor"/>
+				<owl:someValuesFrom rdf:resource="&fibo-der-der-opt;OptionGuarantor"/>
```
Comment (rivettp) [inline DER/DerivativesContracts/Options.rdf:530]: Is this needed? Would it be the market maker or broker?

File: DER/DerivativesContracts/Options.rdf [hunk 6]
```diff
@@ -607,6 +522,52 @@
...
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:label xml:lang="en">vanilla option</rdfs:label>
+		<skos:definition xml:lang="en">common option giving the buyer (holder) the right, but not the obligation, to buy (via a call option) or sell (via a put option) the underlying assets specified at a pre-determined price (i.e., the strike price, fixed or calculated), on or before a specified date (the expiration date)</skos:definition>
+		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">Vanilla options include call or put options that have no special or unusual features, and are typically exchange traded.</fibo-fnd-utl-av:explanatoryNote>
+	</owl:Class>
+	
 	<owl:ObjectProperty rdf:about="&fibo-der-der-opt;hasCalculatedMarketValue">
 		<rdfs:subPropertyOf rdf:resource="&fibo-fnd-acc-cur;hasPrice"/>
 		<rdfs:label xml:lang="en">has calculated market value</rdfs:label>
```
Comment (rivettp) [inline DER/DerivativesContracts/Options.rdf:573]: This is the theoretical value? Should use that term. need to document the calculation.

File: DER/DerivativesContracts/Options.rdf [hunk 7]
```diff
@@ -607,6 +522,52 @@
...
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-agr-ctr;hasContractDuration"/>
+				<owl:someValuesFrom rdf:resource="&fibo-fnd-dt-fd;ExplicitDuration"/>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-agr-ctr;hasCounterparty"/>
```
Comment (rivettp) [inline DER/DerivativesContracts/Options.rdf:547]: is this the buyer? Why not say so?

File: DER/DerivativesContracts/Options.rdf [hunk 8]
```diff
@@ -607,6 +522,52 @@
...
+				<owl:someValuesFrom rdf:resource="&fibo-der-der-opt;TradedOptionsCounterparty"/>
+			</owl:Restriction>
+		</rdfs:subClassOf>
+		<rdfs:subClassOf>
+			<owl:Restriction>
+				<owl:onProperty rdf:resource="&fibo-fnd-pas-pas;hasSeller"/>
```
Comment (rivettp) [inline DER/DerivativesContracts/Options.rdf:553]: distinguish seller (the broker/market maker) from the writer of the option?

**Review by rivettp — APPROVED** [review]
OK for progress but some important questions.

File: DER/DerivativesContracts/Options.rdf (hunk 2 above)
Comment (ElisaKendall) [reply DER/DerivativesContracts/Options.rdf:190]: @rivettp This representation came out of our discussion in the last couple of DER meetings and with back and forth from John G and Jeff B.  We can review again, but that's where this came from.  There are other combination options that need to be subclasses of this class that we don't have covered yet, fyi.

File: DER/DerivativesContracts/Options.rdf (hunk 3 above)
Comment (ElisaKendall) [reply DER/DerivativesContracts/Options.rdf:202]: @rivettp It is true of all equity options regardless of the exchange according to JG - I asked that question explicitly.

File: DER/DerivativesContracts/Options.rdf (hunk 5 above)
Comment (ElisaKendall) [reply DER/DerivativesContracts/Options.rdf:530]: @rivettp We can discuss that - it was in MB's model originally.

File: DER/DerivativesContracts/Options.rdf (hunk 4 above)
Comment (ElisaKendall) [reply DER/DerivativesContracts/Options.rdf:541]: @rivettp For exchange-traded options (vanilla options) apparently this is the case, but we can review that again.

**Pull request merged**